app/controllers/concerns/idv/document_capture_concern.rb

```diff
@@ -81,6 +81,22 @@ def redirect_to_correct_vendor(vendor, in_hybrid_mobile)
       redirect_to correct_path
     end
 
+    def track_document_request_event(document_request:, document_response:, timer:)
+      document_request_body = JSON.parse(document_request.body, symbolize_names: true)[:config]
+      log_extras = {
+        reference_id: document_response.to_h[:referenceId],
+        vendor: 'Socure',
+        vendor_request_time_in_ms: timer.results['vendor_request'],
+        success: @url.present?,
+        document_type: document_request_body[:documentType],
+      }
+      analytics_hash = log_extras.merge(analytics_arguments).
+        merge(document_request_body).except(
+          :documentType, # requested document type
+        ).merge(response_body: document_response.to_h)
```

Comment:
@amirbey I think the except for : documentType is needed on line 95, since even if document_type is defined in responseBody, documentType (camelCase) is being returned as a part of the request_body[:config]. Since the track event is expecting document_type (snake case), we need to get rid of it. Also, just pulled from main. I am not seeing documentType in responseBody. I see document_type as part of the extra attributes of doc_v_result_response, but I don't think that is applicable here. Happy to discuss further, but in short I think we need line 94-96 the way it is now

Reply:
fair enough 👍🏿